tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -6841,12 +6845,43 @@ Status ConvertGraphDefToEngine(
   TF_RETURN_IF_ERROR(statusor.status());
   std::unique_ptr<Converter> converter = std::move(statusor.ValueOrDie());
 
+  GraphDef graph = gdef;
+  if (cluster != nullptr) {
+    bool apply_layout_optim;
+    Status status =
+        ReadBoolFromEnvVar("TF_TRT_ENABLE_LAYOUT_OPTIMIZER",
+                           /*default_value=*/true, &apply_layout_optim);
+    if (!status.ok()) {
+      LOG(ERROR) << status;
+    }
+    if (apply_layout_optim) {
+      tensorflow::grappler::GrapplerItem grappler_item;
+      grappler_item.graph = gdef;
+      // TensorRT API requires the input for convolution to be in NCHW.
+      tensorflow::grappler::GenericLayoutOptimizer layout_optimizer("NCHW");
+      layout_optimizer.Optimize(cluster, grappler_item, &graph);
+
+      grappler_item.graph = graph;
+
+      tensorflow::grappler::ConstantFolding const_optimizer(nullptr,
+                            /*disable_compressed_tensor_optimization=*/false,
+                            /*fold_quantization_emulation=*/false);
+      const_optimizer.Optimize(cluster, grappler_item, &graph);
```

Comment:
Siimlar to the above, need to handle the return value from the call.

Reply:
Done